packages/components/fields/async-creatable-select-field/src/async-creatable-select-field.tsx

```diff
@@ -467,7 +489,7 @@ export default class AsyncCreatableSelectField extends Component<
             isOptionDisabled={this.props.isOptionDisabled}
             isMulti={this.props.isMulti}
             isSearchable={this.props.isSearchable}
-            hasWarning={this.props.hasWarning}
+            hasWarning={hasWarning}
```

Comment:
I'm a bit confused about one aspect in general. Based on my understanding, the original requirements were outlined [here](https://github.com/commercetools/ui-kit/issues/2671#issuecomment-1886715030).

I read [this comment](https://github.com/commercetools/ui-kit/pull/2698#issuecomment-1905713921) but I'm not sure if this was addressed. 

Previously, the `hasWarning` prop alone triggered the border display, but it's no longer considered. Is the `hasWarning` prop deprecated? Shouldn't this be mentioned in the changeset?

Reply:
That's a good catch, Kacper.

I think the `hasWarning` prop we currently have should be deprecated, but that means keeping it so consumers have time to transition from one prop (`hasWarning`) to the other (`warnings`).

We probably need the `hasWarning` constant calculation to be adjusted:
```js
const hasWarning =
     this.props.hasWarnings ||
      (AsyncCreatableSelectInput.isTouched(this.props.touched) &&
      hasWarnings(this.props.warnings));
```

And also update the jsdoc of that property to mention about it.

@kark what do you think?

Reply:
Ahhh yeahh I think it should. @CarlosCortizasCT Whats your take on this?

Reply:
Cool stuff. @CarlosCortizasCT So the jsdoc should mention that the field is already deprecated or is their a specific copy you would like to use?

Reply:
Hi @obulaworld 

> @CarlosCortizasCT So the jsdoc should mention that the field is already deprecated or is their a specific copy you would like to use?

I mean both.
For example:
```
  /**
   * Indicates the input field has a warning.
   * @deprecated Please use the `warnings` prop instead so users know the reason why the field is in warning state.
   */
  hasWarning?: boolean;
```

> Also does this mean we keep the hasWarning boolean prop in the story?

We should remove it as we don't want consumers to use it anymore.

Reply:
Done